onnx_chainer/functions/pooling.py

```diff
@@ -3,24 +3,28 @@
 
 def convert_AveragePooling2D(
         func, onnx_op_name, opset_version, input_names, output_names, parameters):
+    # Supports cover_all by setting extra padding
+    if func.cover_all:
+        dph, dpw = func.sy - 1, func.sx - 1
+    else:
+        dph, dpw = 0, 0
     if opset_version == 1 or opset_version == 7:
         return helper.make_node(
             onnx_op_name, input_names, output_names,
             kernel_shape=(func.kh, func.kw),
-            pads=(func.ph, func.pw, func.ph, func.pw),
+            pads=(func.ph, func.pw, func.ph + dph, func.pw + dpw),
             strides=(func.sy, func.sx)
         ),
 
 
 def convert_AveragePoolingND(func, onnx_op_name, opset_version, input_names, output_names, parameters):
-    pad = []
-    x_ndim = len(func.inputs[0].shape)
-    k_ndim = len(func.ksize)
-    for _ in range(x_ndim - k_ndim):
-        pad.append(0)
```

Comment:
Why did you remove this part? I think Chainer doesn't have a way to pad along batch dimension or channel dimension, but the ONNX operator `AveragePool` can represent such case. So I just filled with the dimensions other than the spatial dimensions.

Reply:
> but the ONNX operator AveragePool can represent such case.

? Sorry, but at least it seems that `AveragePool` can accept `pad` without filling the batch and the channel dimensions, and as far as we refer to the examples of AveragePool [1], the approach without batch and channel is valid.

[1] https://github.com/onnx/onnx/blob/master/docs/Operators.md#AveragePool

Reply:
Oops, sorry I was wrongly taking the non-spatial dimensions into account when converting pooling functions... Now it's clear!